File: nestedclass.cs
using System;

class GrandFather {
    public string name;
    public int age;

    public GrandFather(string name, int age) {
        this.name = name;
        this.age = age;
    }

    public GrandFather() : this("GrandFather", 70) {}

    public void infoFamilyGrandFather() {
        Console.WriteLine(name + " " + age);
    }

    public class Uncle {
        public string name;
        public int age;

        public Uncle(string name, int age) {
            this.name = name;
            this.age = age;
        }

        public Uncle() : this("Uncle", 45) {}

        public void infoFamilyUncle() {
            Console.WriteLine(name + " " + age);
        }
    }

    public class Father {
        public string name;
        public int age;

        public Father(string name, int age) {
            this.name = name;
            this.age = age;
        }

        public Father() : this("Father", 40) {}

        public void infoFamilyFather() {
            Console.WriteLine(name + " " + age);
        }

        public class Brother {
            public string name; 
            public int age;

            public Brother(string name, int age) {
                this.name = name;
                this.age = age;
            }

            public Brother() : this("Brother", 18) {}

            public void infoFamilyBrother() {
                Console.WriteLine(name + " " + age);
            }

            public class Niece {
                public string name; 
                public int age;

                public Niece(string name, int age) {
                    this.name = name;
                    this.age = age;
                }

                public Niece() : this("Niece", 3) {}

                public void infoFamilyNiece() {
                    Console.WriteLine(name + " " + age);
                }
            }
        }

        public class Me {
            public string name; 
            public int age;

            public Me(string name, int age) {
                this.name = name;
                this.age = age;
            }

            public Me() : this("Me", 20) {}

            public void infoFamilyMe() {
                Console.WriteLine(name + " " + age);
            }

            public class MyDog {
                public string name; 
                public int age;

                public MyDog(string name, int age) {
                    this.name = name;
                    this.age = age;
                }

                public MyDog() : this("Dog", 2) {}

                public void infoDog() {
                    Console.WriteLine(name + " " + age);
                }
            }
        }
    }
}

class Program {
    static void Main() {
        GrandFather gf = new GrandFather();
        gf.infoFamilyGrandFather();

        GrandFather.Uncle u = new GrandFather.Uncle();
        u.infoFamilyUncle();

        GrandFather.Father f = new GrandFather.Father();
        f.infoFamilyFather();

        GrandFather.Father.Brother b = new GrandFather.Father.Brother();
        b.infoFamilyBrother();

        GrandFather.Father.Brother.Niece n = new GrandFather.Father.Brother.Niece();
        n.infoFamilyNiece();

        GrandFather.Father.Me me = new GrandFather.Father.Me();
        me.infoFamilyMe();

        GrandFather.Father.Me.MyDog d = new GrandFather.Father.Me.MyDog();
        d.infoDog();
    }
}

File: Nestedclass.py
class GrandFather:
    def __init__(self, name="GrandFather", age=70):
        self.name = name
        self.age = age

    def infoFamilyGrandFather(self):
        print(self.name, self.age)

    class Uncle:
        def __init__(self, name="Uncle", age=45):
            self.name = name
            self.age = age

        def infoFamilyUncle(self):
            print(self.name, self.age)

    class Father:
        def __init__(self, name="Father", age=40):
            self.name = name
            self.age = age

        def infoFamilyFather(self):
            print(self.name, self.age)

        class Brother:
            def __init__(self, name="Brother", age=18):
                self.name = name
                self.age = age

            def infoFamilyBrother(self):
                print(self.name, self.age)

            class Niece:
                def __init__(self, name="Niece", age=3):
                    self.name = name
                    self.age = age

                def infoFamilyNiece(self):
                    print(self.name, self.age)

        class Me:
            def __init__(self, name="Me", age=20):
                self.name = name
                self.age = age

            def infoFamilyMe(self):
                print(self.name, self.age)

            class MyDog:
                def __init__(self, name="Dog", age=2):
                    self.name = name
                    self.age = age

                def infoDog(self):
                    print(self.name, self.age)


gf = GrandFather()
gf.infoFamilyGrandFather()

GrandFather.Uncle().infoFamilyUncle()
GrandFather.Father().infoFamilyFather()
GrandFather.Father.Brother().infoFamilyBrother()
GrandFather.Father.Brother.Niece().infoFamilyNiece()
GrandFather.Father.Me().infoFamilyMe()
GrandFather.Father.Me.MyDog().infoDog()
